### PR TITLE
Pamela454 seed some data 2

### PR DESCRIPTION
**Pull request by Pamela454**
Seeded database with sample data. 

## Dev Summary

[Mocks](link)

    

In the future, could also user faker data as seen in this example:
 https://dev.to/franciscomendes10866/how-to-seed-database-using-typeorm-seeding-4kd5

## Test Plan

repro steps:
1. server run npm run seed:drop to drop all data in database and seed with sample data 
2. sample data is seeded into tables for users, transactions, organizations, categories, assets, messages, user_organizations

Screen Shot 2022-05-19 at 3 47 27 PM

Screen Shot 2022-05-19 at 3 47 33 PM

## Resources

- [stackoverflow is cool](stackoverflow.com)

### CODE REVIEW

File: server/src/seeder/seeder.module.ts [hunk 1]
```diff
@@ -32,9 +45,16 @@ export class SeederModule implements OnApplicationBootstrap {
...
 
     // remove all data
-    this.seederService.truncateFromAllTables();
+    //this.seederService.truncateFromAllTables();
 
     // seed the users table
-    await this.seederService.seedUsersAsync();
+    this.seederService
+      .seedUsersAsync()
+      .then(() => this.seederService.seedAssetsAsync())
```
Comment (jd2rogers2) [inline server/src/seeder/seeder.module.ts:53]: in terms of creating the relationships, i think if we make these async seeding service functions return the data that we get back from the `.create` functions, we should be able to access that data in the next `.then`'s callback
aka:
we return the new user in `seedUsersAsync`
line above changes to `.then((newUser) => this.seederService.seedAssetsAsync(newUser))`
and then creating the relationship with that user is easy
hope that works!

File: server/src/database-connection.service.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export class DatabaseConnectionService implements TypeOrmOptionsFactory {
       password: process.env.DATABASE_PASSWORD,
       database: process.env.DATABASE_DB,
       synchronize: true, //shouldn't be used in production
-      dropSchema: false, //toggle to true to clear database schema
+      dropSchema: true, //toggle to true to clear database schema
```
[the remark this answers is not in the document]
Comment (Pamela454) [reply server/src/database-connection.service.ts:16]: Corrected. Also, I had left the call to truncateFromAllTables() commented out as well.  I have now removed it. Allowing the schema to drop and calling truncateFromAllTables() together was locking the database.

File: server/src/seeder/seeder.module.ts (hunk 1 above)
Comment (Pamela454) [reply server/src/seeder/seeder.module.ts:53]: updated to pass results from previous promise.

**Review by jd2rogers2 — APPROVED** [review]
one nit then land this sucker! this is awesome!!!

File: server/src/assets/entities/asset.entity.ts [hunk 3]
```diff
@@ -56,7 +56,7 @@ export class Asset {
   imgUrls: string[];
 
   @ManyToOne(() => User, (user) => user.assets, { eager: true })
-  @JoinColumn({ name: 'poster_id' })
+  @JoinColumn({ name: 'posterid' })
```
[the remark this answers is not in the document]
Comment (Pamela454) [reply server/src/assets/entities/asset.entity.ts]: ok completed. plus made truncateFromAllTables asynchronous and wait for completion before seeding to correct database locking

**Pull request merged**